sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -1029,4 +1031,43 @@ object JdbcUtils extends Logging {
       statement.close()
     }
   }
+
+  def executeQuery(conn: Connection, options: JDBCOptions, sql: String): ResultSet = {
+    val statement = conn.createStatement
+    try {
+      statement.setQueryTimeout(options.queryTimeout)
+      statement.executeQuery(sql)
```

Comment:
Don't we need to handle `SQLException`?

Reply:
?

Reply:
+1 for @viirya 's comment.

Reply:
Sorry, somehow I missed this comment. Fixed.